src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7839,4 +7839,28 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_ParamsCollections" xml:space="preserve">
     <value>params collections</value>
   </data>
+  <data name="ERR_CantInferMethTypeArgs_DynamicArgumentWithParamsCollections" xml:space="preserve">
+    <value>The type arguments for method '{0}' cannot be inferred from the usage because an argument with dynamic type is used and the method has a non-array params collection parameter. Try specifying the type arguments explicitly.</value>
+  </data>
+  <data name="ERR_ParamsCollectionAmbiguousDynamicArgument" xml:space="preserve">
+    <value>Ambiguity between expanded and normal forms of non-array params collection parameter of '{0}', the only corresponding argument has dynamic type. Consider casting the dynamic argument.</value>
+  </data>
+  <data name="WRN_DynamicDispatchToParamsCollectionMethod" xml:space="preserve">
+    <value>One or more overloads of method '{0}' having non-array params collection parameter might be applicable only in expanded form which is not supported during dynamic dispatch.</value>
+  </data>
+  <data name="WRN_DynamicDispatchToParamsCollectionMethod_Title" xml:space="preserve">
+    <value>One or more overloads of method having non-array params collection parameter might be applicable only in expanded form which is not supported during dynamic dispatch.</value>
+  </data>
+  <data name="WRN_DynamicDispatchToParamsCollectionIndexer" xml:space="preserve">
+    <value>One or more overloads of indexer having non-array params collection parameter might be applicable only in expanded form which is not supported during dynamic dispatch.</value>
+  </data>
+  <data name="WRN_DynamicDispatchToParamsCollectionIndexer_Title" xml:space="preserve">
+    <value>One or more overloads of indexer having non-array params collection parameter might be applicable only in expanded form which is not supported during dynamic dispatch.</value>
+  </data>
+  <data name="WRN_DynamicDispatchToParamsCollectionConstructor" xml:space="preserve">
+    <value>One or more overloads of constructor having non-array params collection parameter might be applicable only in expanded form which is not supported during dynamic dispatch.</value>
```

Comment:
For this and the indexer warning, consider putting the type name in the error message.

Additionally, both `constructor` and `indexer` need to be pluralized here, or have the `(s)` suffix to indicate that they may be pluralized.